js/src/hooks/useCreateNoticeForRejectedPromises.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Internal dependencies
+ */
+import useCreateNoticeForMultipleErrors from '.~/hooks/useCreateNoticeForMultipleErrors';
+
+/**
+ *
+ * @callback CreateNoticeForRejectedPromises
+ * @param  {Array<Promise>} promises Promises to be called.
+ * @param  {Array<string>} promiseNames Name of each provided promise in the first argument. It will be used to create the notice and needs to follow the same order that in the first argument.
+ * @return {Array<string>} The error messages that have been displayed.
+ */
+
+/**
+ * A hook to create an error notice for rejected promises.
+ *
+ * @return {CreateNoticeForRejectedPromises} - A function that will create the error notice for the rejected promises.
+ */
+export default function useCreateNoticeForRejectedPromises() {
+	const {
+		createNoticeForMultipleErrors,
+	} = useCreateNoticeForMultipleErrors();
+
+	/**
+	 * @type {CreateNoticeForRejectedPromises} CreateNoticeForRejectedPromises	 *
+	 */
+	const createNoticeForRejectedPromises = async (
+		promises,
+		promiseNames
+	) => {
+		const results = await Promise.allSettled( promises );
+		const rejectedMessages = results.reduce( ( acc, item, index ) => {
+			if ( item.status === 'rejected' && promiseNames[ index ] ) {
+				acc.push( promiseNames[ index ] );
+			}
+			return acc;
+		}, [] );
+
+		const isPartiallySuccessful = rejectedMessages.length < promises.length;
```

Comment:
🚧 
When all promises are rejected, it will make the `useCreateNoticeForMultipleErrors` show an error notice with "Other changes have been saved." in its text.

Reply:
Is it not because of this?

>This PR is partially closing https://github.com/woocommerce/google-listings-and-ads/issues/1331 as I realized that saveShippingTimes does not propagate the error because upsertShippingTimes catches the error. I will work on this in a different PR.

I can't reproduce it 🤔 , as far as I can see `isPartiallySuccessful` will be `false` if `rejectedMessages.length` has the same length that the promises array.

Reply:
Oh, sorry. 🙏 🤦‍♂️ 

I thought "Would it be a problem if pass `isPartiallySuccessful` as `true` to call `createNoticeForMultipleErrors` when all promises are resolved?" when viewing this line of code. But I got the logic upside down when writing the comment after switching back and forth several files.

💅 
And I learned that `useCreateNoticeForMultipleErrors` would check the length of `errorMessages`, so it won't be a problem. But I'm still unsure about the original comment I was supposed to ask - Should `isPartiallySuccessful` be `true` to match its semantic when all promises are resolved? Or maybe it would be clearer with:
```js
// `useCreateNoticeForRejectedPromises` - Move the length check to here and rename isPartiallySuccessful to isAllFailed or something more semantically.
if ( rejectedMessages.length ) {
	const isAllFailed = rejectedMessages.length === promises.length;
	createNoticeForMultipleErrors( rejectedMessages, isAllFailed );
}

// `useCreateNoticeForMultipleErrors` - Corresponding changes to above.
const createNoticeForMultipleErrors = (
	errorMessages,
	isAllFailed = true
) => {
	const listErrors = concatenateListOfWords( errorMessages );

	const content = sprintf(
		__(
			'%1$s. %2$s Please try again later.',
			'google-listings-and-ads'
		),
		listErrors,
		isAllFailed ? '' : 'Other changes have been saved.'
	);

	createNotice( 'error', content );
};
```